src/include/platform/ConnectivityManager.h

```diff
@@ -177,19 +177,28 @@ class ConnectivityManager
     void ClearWiFiStationProvision();
     CHIP_ERROR GetAndLogWiFiStatsCounters();
 #if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
-    struct WiFiPAFAdvertiseParam;
+    struct WiFiPAFAdvertiseParam
+    {
+        /* Frequency list */
+        uint16_t freq_list_len;
+        std::unique_ptr<uint16_t[]> freq_list;
```

Comment:
Thinking aloud, maybe we should make https://github.com/project-chip/connectedhomeip/blob/master/src/app/data-model-provider/MetadataList.h#L123 more public than `data-model-provider` since it has a few things that may be desirable:
  - ensure read-only (seems desirable here)
  - packages data & length together span-like
  - uses Platform::Memory instead of plain heap.
  
  The last one is what caught my eye: we generally try to avoid heap in generic headers and this seems generic.
  
  I see this is moving existing code .... however maybe we should not have used heap to begin with in super-generic code.

Reply:
Could you let me know how to make it more public? I got this error while using gn check.
Thank you

$ gn check --check-generated out/aarch64/
ERROR at //third_party/connectedhomeip/src/include/platform/ConnectivityManager.h:27:11: Include not allowed.
#include <app/data-model-provider/MetadataList.h>
          ^-------------------------------------
It is not in any dependency of
  //third_party/connectedhomeip/src/platform:platform
The include file is in the target(s):
  //third_party/connectedhomeip/src/app/data-model-provider:data-model-provider
which should somehow be reachable.
This might be a false alarm if you are using conditional include. See "gn help nogncheck".

Reply:
created #38364 however that is a larger rename so I imagine we do not want to wait for it to merge and block this PR ... could we add a followup issue?

Reply:
The code to use ReadOnlyBuffer has been added. Please review.
Thank you